<ui version="4.0" >
 <author>Gary Sherman</author>
 <comment>Query builder for PostgreSQL layers (using PostGIS)	</comment>
 <class>QgsPgQueryBuilderBase</class>
 <widget class="QDialog" name="QgsPgQueryBuilderBase" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>644</width>
    <height>533</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy>
    <hsizetype>7</hsizetype>
    <vsizetype>7</vsizetype>
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle" >
   <string>PostgreSQL Query Builder</string>
  </property>
  <property name="windowIcon" >
   <iconset/>
  </property>
  <property name="modal" >
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" >
   <property name="margin" >
    <number>10</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item row="4" column="0" colspan="2" >
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <spacer>
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType" >
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" >
        <size>
         <width>30</width>
         <height>21</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnClear" >
       <property name="text" >
        <string>Clear</string>
       </property>
       <property name="default" >
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnTest" >
       <property name="text" >
        <string>Test</string>
       </property>
       <property name="default" >
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnOk" >
       <property name="text" >
        <string>Ok</string>
       </property>
       <property name="default" >
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Cancel" >
       <property name="text" >
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType" >
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" >
        <size>
         <width>30</width>
         <height>21</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="1" column="1" >
    <widget class="Q3GroupBox" name="groupBox2" >
     <property name="sizePolicy" >
      <sizepolicy>
       <hsizetype>7</hsizetype>
       <vsizetype>7</vsizetype>
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title" >
      <string>Values</string>
     </property>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>10</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="1" column="1" >
       <widget class="QPushButton" name="btnGetAllValues" >
        <property name="text" >
         <string>All</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2" >
       <widget class="Q3ListBox" name="lstValues" />
      </item>
      <item row="1" column="0" >
       <widget class="QPushButton" name="btnSampleValues" >
        <property name="text" >
         <string>Sample</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" >
    <widget class="Q3GroupBox" name="groupBox1" >
     <property name="title" >
      <string>Fields</string>
     </property>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>10</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="0" column="0" >
       <widget class="Q3ListBox" name="lstFields" />
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2" >
    <widget class="QLabel" name="lblDataUri" >
     <property name="text" >
      <string>Datasource:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2" >
    <widget class="Q3GroupBox" name="groupBox4" >
     <property name="title" >
      <string>Operators</string>
     </property>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>10</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="0" column="0" >
       <widget class="QPushButton" name="btnEqual" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>=</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5" >
       <widget class="QPushButton" name="btnIn" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>IN</string>
        </property>
       </widget>
      </item>
      <item row="0" column="6" >
       <widget class="QPushButton" name="btnNotIn" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>NOT IN</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="QPushButton" name="btnLessThan" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>&lt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" >
       <widget class="QPushButton" name="btnGreaterThan" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>></string>
        </property>
       </widget>
      </item>
      <item row="0" column="4" >
       <widget class="QPushButton" name="btnPct" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>%</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QPushButton" name="btnLessEqual" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>&lt;=</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="QPushButton" name="btnGreaterEqual" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>>=</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QPushButton" name="btnNotEqual" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>!=</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3" >
       <widget class="QPushButton" name="btnLike" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>LIKE</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4" >
       <widget class="QPushButton" name="btnAnd" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>AND</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3" >
       <widget class="QPushButton" name="btnILike" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>ILIKE</string>
        </property>
       </widget>
      </item>
      <item row="1" column="5" >
       <widget class="QPushButton" name="btnOr" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>OR</string>
        </property>
       </widget>
      </item>
      <item row="1" column="6" >
       <widget class="QPushButton" name="btnNot" >
        <property name="maximumSize" >
         <size>
          <width>74</width>
          <height>32767</height>
         </size>
        </property>
        <property name="text" >
         <string>NOT</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2" >
    <widget class="Q3GroupBox" name="groupBox3" >
     <property name="maximumSize" >
      <size>
       <width>32767</width>
       <height>116</height>
      </size>
     </property>
     <property name="title" >
      <string>SQL where clause</string>
     </property>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>10</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="0" column="0" >
       <widget class="Q3TextEdit" name="txtSQL" />
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <customwidgets>
  <customwidget>
   <class>Q3GroupBox</class>
   <extends>QGroupBox</extends>
   <header>Qt3Support/Q3GroupBox</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Q3ListBox</class>
   <extends>Q3Frame</extends>
   <header>q3listbox.h</header>
  </customwidget>
  <customwidget>
   <class>Q3TextEdit</class>
   <extends>Q3Frame</extends>
   <header>q3textedit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>btnSampleValues</tabstop>
  <tabstop>btnGetAllValues</tabstop>
  <tabstop>btnEqual</tabstop>
  <tabstop>btnLessThan</tabstop>
  <tabstop>btnGreaterThan</tabstop>
  <tabstop>btnLike</tabstop>
  <tabstop>btnPct</tabstop>
  <tabstop>btnIn</tabstop>
  <tabstop>btnNotIn</tabstop>
  <tabstop>btnLessEqual</tabstop>
  <tabstop>btnGreaterEqual</tabstop>
  <tabstop>btnNotEqual</tabstop>
  <tabstop>btnILike</tabstop>
  <tabstop>btnAnd</tabstop>
  <tabstop>btnOr</tabstop>
  <tabstop>btnNot</tabstop>
  <tabstop>btnClear</tabstop>
  <tabstop>btnTest</tabstop>
  <tabstop>btnOk</tabstop>
  <tabstop>Cancel</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>Cancel</sender>
   <signal>clicked()</signal>
   <receiver>QgsPgQueryBuilderBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>417</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel" >
     <x>501</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
